domain/schema/model/sql/0020-network.sql

```diff
@@ -23,3 +23,163 @@ CREATE TABLE cloud_container (
     FOREIGN KEY (net_node_uuid)
     REFERENCES net_node (uuid)
 );
+
+CREATE TABLE cloud_container_ip_address (
+    net_node_uuid TEXT NOT NULL,
+    ip_address_uuid TEXT NOT NULL,
+    CONSTRAINT fk_cloud_container_ip_address_net_node_uuid
+    FOREIGN KEY (net_node_uuid)
+    REFERENCES cloud_container (net_node_uuid),
+    CONSTRAINT fk_cloud_container_ip_address_ip_address_uuid
+    FOREIGN KEY (ip_address_uuid)
+    REFERENCES ip_address (uuid),
+    PRIMARY KEY (net_node_uuid, ip_address_uuid)
+);
+
+CREATE TABLE cloud_container_port (
+    net_node_uuid TEXT NOT NULL,
+    port TEXT NOT NULL,
+    CONSTRAINT fk_cloud_container_port_net_node_uuid
+    FOREIGN KEY (net_node_uuid)
+    REFERENCES cloud_container (net_node_uuid),
+    PRIMARY KEY (net_node_uuid, port)
+);
+
+-- ip_address_type represents the possible ways of specifying
+-- an address, either a hostname resolvable by dns lookup,
+-- or IPv4 or IPv6 address.
+CREATE TABLE ip_address_type (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_ip_address_type_name
+ON ip_address_type (name);
+
+INSERT INTO ip_address_type VALUES
+(0, 'hostname'),
```

Comment:
This reflects what we have in core network. It needs to match.
```
// AddressType represents the possible ways of specifying a machine location by
// either a hostname resolvable by dns lookup, or IPv4 or IPv6 address.
type AddressType string

const (
	HostName    AddressType = "hostname"
	IPv4Address AddressType = "ipv4"
	IPv6Address AddressType = "ipv6"
)
```

Reply:
I will say _from_ the outset we've said that the data should be correct _at the cost_ of the code. 

I think this could be solved with a view on to multiple tables that __are__ correctly modeled.

Reply:
This logic is ingrained in a lot of places in core network. 
certificate updater
```
func (c *CertificateUpdater) updateCertificate(addresses network.SpaceAddresses) error {
	c.logger.Debugf("new machine addresses: %#v", addresses)
	c.addresses = addresses

	request := c.authority.LeafRequestForGroup(pki.ControllerIPLeafGroup)

	for _, addr := range addresses {
		if addr.Value == "localhost" {
			continue
		}

		switch addr.Type {
		case network.HostName:
			request.AddDNSNames(addr.Value)
```
manual provider host address
```
// HostAddress returns an network.Address for the specified
// hostname, depending on whether it is an IP or a resolvable
// hostname. The address is given public scope.
func HostAddress(hostname string) (network.ProviderAddress, error) {
...
	// Only a resolvable hostname may be used as a public address.
	if _, err := netLookupHost(hostname); err != nil {
		return network.ProviderAddress{}, err
	}
	addr := network.ProviderAddress{
		MachineAddress: network.MachineAddress{
			Value: hostname,
			Type:  network.HostName,
			Scope: network.ScopePublic,
		},
```
And others.... the point is there's some fundamental code paths affected.

If we want to make progress in getting off mongo, we must do certain bits of the uplift in stages. We can retain the currently modelled address type enums in dqlite, and that bit of mongo can be deleted. Then in a safe way we can redo this aspect of the network model. ie in order
1. mongo->dqlite translation
2. retarget existing code to dqlite
3. fix network model when it's all just dqlite

If we try and intermingle those steps it won't end well.

Reply:
As a way forward, I've used separate tables for ip address vs fqdn address and a view to present them to the existing domain logic